react/react-training/src/app/components/AppRoutes/components/AuthRoute/index.js

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { Redirect, Route } from 'react-router';
+import PropTypes from 'prop-types';
+
+function AuthComponent({ component, exact, path, userIsLogged }) {
+  if (userIsLogged) {
+    return <Redirect to="/game" />;
+  }
+  return <Route exact={exact} path={path} component={component} />;
+}
+
+AuthComponent.propTypes = {
+  component: PropTypes.func.isRequired,
```

Comment:
Change this proptype for `element`

Reply:
Rejected this recommendation because appears this error:
`Warning: Failed prop type: Invalid prop 'component' of type 'function' supplied to 'AuthComponent', expected a single ReactElement.`